src/main/java/io/vertx/core/parsetools/impl/LengthFieldParserImpl.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.parsetools.impl;
+
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.impl.Arguments;
+import io.vertx.core.parsetools.LengthFieldParser;
+import io.vertx.core.parsetools.RecordParser;
+import io.vertx.core.streams.ReadStream;
+
+/**
+ * @author <a href="mailto:emad.albloushi@gmail.com">Emad Alblueshi</a>
+ */
+
+public class LengthFieldParserImpl implements LengthFieldParser {
+
+  private Handler<Throwable> exceptionHandler;
+  private final RecordParser parser;
+
+  private long frameLength = -1;
+  private final int length;
+  private final int offset;
```

Comment:
The purpose of the offset is to skip any field in the frame that precedes the length field. But after the length has been read, the user might want to know what was the value of the skipped field.

Looking at the implementation below, it seems the user will only get fields that come after the length fields.

The user should be able to tell if they want to receive the whole frame (including the length field) or only the fields that come after the length-field. They could do it with a boolean param provided when creating this parser.

Reply:
`skip = true`
``` 
 BEFORE PARSING (14 bytes)                     AFTER PARSING (12 bytes)
 +----------+----------+----------------+      +----------+-----+
 | offset   |  Length  | Actual Content |----->| Actual Content |
 | byte(1)  | byte(1)  | "HELLO, WORLD" |      | "HELLO, WORLD" |
 +----------+----------+----------------+      +----------+-----+
```
`skip = false`

```
 BEFORE PARSING (14 bytes)                      AFTER PARSING (14 bytes)
 +----------+----------+----------------+      +----------+----------+----------------+
 |  offset  |  Length  | Actual Content |----->|  offset  |  Length  | Actual Content |
 |  byte(1) | byte(1)  | "HELLO, WORLD" |      |  byte(1) | byte(1)  | "HELLO, WORLD" |
 +----------+----------+----------------+      +----------+----------+----------------+
```

@tsegismont right?

Reply:
The first one is already covered if we simply provide a boolean ’skip = true’ 

but if ‘skip = false’ I need to know to add it asap.

Reply:
I think we need to support the first two cases.

Reply:
@tsegismont Done